src/Kafka/ManageKafkaPermissions/components/ResourceTypeOperationFilter/ResourceTypeOperationFilter.tsx

```diff
@@ -233,7 +233,7 @@ export const ResourceTypeOperationFilter: React.VFC<
   const filterItems = (
     item: TreeViewDataItem,
     checkedItem: TreeViewDataItem
-  ) => {
+  ): TreeViewDataItem | boolean | undefined => {
```

Comment:
This type is a bit confusing, wouldn't `TreeViewDataItem | boolean` be enough, and return `false` instead of `undefined` be better?